src/Umbraco.Cms.ManagementApi/Controllers/ExamineManagement/RebuildExamineManagementController.cs

```diff
@@ -0,0 +1,115 @@
+    public RebuildExamineManagementController(
+        ILogger<RebuildExamineManagementController> logger,
+        IIndexRebuilder indexRebuilder,
+        ITemporaryIndexingService temporaryIndexingService,
+        IExamineManager examineManager)
+    {
+        _logger = logger;
+        _indexRebuilder = indexRebuilder;
+        _temporaryIndexingService = temporaryIndexingService;
+        _examineManager = examineManager;
+    }
+
+    /// <summary>
+    ///     Rebuilds the index
+    /// </summary>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    [HttpPost("rebuild")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(OkResult), StatusCodes.Status200OK)]
+    public async Task<IActionResult> Rebuild(string indexName)
+    {
+        if (!_examineManager.TryGetIndex(indexName, out var index))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Index Not Found",
+                Detail = $"No index found with name = {indexName}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
+
+            return await Task.FromResult(BadRequest(invalidModelProblem));
+        }
+
+        if (!_indexRebuilder.CanRebuild(index.Name))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Could not validate the populator",
+                Detail = $"The index {index?.Name} cannot be rebuilt because we could not validate its associated {typeof(IIndexPopulator)}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
+
+            return await Task.FromResult(BadRequest(invalidModelProblem));
+        }
+
+        _logger.LogInformation("Rebuilding index '{IndexName}'", indexName);
+
+        // remove it in case there's a handler there already
+        index!.IndexOperationComplete -= Indexer_IndexOperationComplete;
+
+        //now add a single handler
+        index.IndexOperationComplete += Indexer_IndexOperationComplete;
+
+        try
+        {
+            _temporaryIndexingService.Set(indexName);
+            _indexRebuilder.RebuildIndex(indexName);
+
+            return await Task.FromResult(Ok());
+        }
+        catch (Exception ex)
+        {
+            //ensure it's not listening
+            index.IndexOperationComplete -= Indexer_IndexOperationComplete;
+            _logger.LogError(ex, "An error occurred rebuilding index");
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Index cannot be rebuilt",
+                Detail = $"The index {index.Name} cannot be rebuilt because it does not have an associated {typeof(IIndexPopulator)}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
```

Comment:
At this point the index *will* have an index populator registered, otherwise this action would have failed earlier. I suggest referring to the log for details.

```suggestion
            var invalidModelProblem = new ProblemDetails
            {
                Title = "Index could not be rebuilt",
                Detail = $"The index {index.Name} could not be rebuild. Check the log for details on this error.",
                Status = StatusCodes.Status400BadRequest,
                Type = "Error",
            };
```